precompiles/assets/assets.go

```diff
@@ -183,6 +178,16 @@ func (p Precompile) Run(evm *vm.EVM, contract *vm.Contract, readOnly bool) (bz [
 		return nil, vm.ErrOutOfGas
 	}
 
+	if p.IsTransaction(method.Name) {
```

Comment:
this `isTransaction` is not safe: if some newly added method would write the store but is forgotten to be declared in `p.IsTransaction`, this function would return false. Given `isTransaction` is enforced by precompile interface, could we add a new function like `safeIsTransaction` that would return an error if it is not declared?

Reply:
Changed `IsTransaction` to `panic` in the `default` situation and captured this `panic` in `init` to force developers to remember this.